src/handler/http/request/alerts/mod.rs

```diff
@@ -13,6 +13,39 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+use actix_web::HttpResponse;
+
+use crate::{
+    common::meta::http::HttpResponse as MetaHttpResponse, service::alerts::alert::AlertError,
+};
+
 pub mod alert;
 pub mod destinations;
 pub mod templates;
+
+impl From<AlertError> for HttpResponse {
+    fn from(value: AlertError) -> Self {
+        match &value {
+            AlertError::InfraError(err) => MetaHttpResponse::internal_error(err),
+            AlertError::CreateDefaultFolderError => MetaHttpResponse::internal_error(value),
+            AlertError::AlertNameMissing => MetaHttpResponse::bad_request(value),
+            AlertError::AlertNameOfgaUnsupported => MetaHttpResponse::bad_request(value),
+            AlertError::AlertNameContainsForwardSlash => MetaHttpResponse::bad_request(value),
+            AlertError::AlertDestinationMissing => MetaHttpResponse::bad_request(value),
+            AlertError::CreateAlreadyExists => MetaHttpResponse::conflict(value),
+            AlertError::AlertNotFound => MetaHttpResponse::not_found(value),
+            AlertError::AlertDestinationNotFound { .. } => MetaHttpResponse::not_found(value),
+            AlertError::StreamNotFound { .. } => MetaHttpResponse::not_found(value),
+            AlertError::DecodeVrl(err) => MetaHttpResponse::bad_request(err),
+            AlertError::ParseCron(err) => MetaHttpResponse::bad_request(err),
+            AlertError::RealtimeMissingCustomQuery => MetaHttpResponse::bad_request(value),
+            AlertError::SqlMissingQuery => MetaHttpResponse::bad_request(value),
+            AlertError::SqlContainsSelectStar => MetaHttpResponse::bad_request(value),
+            AlertError::PromqlMissingQuery => MetaHttpResponse::bad_request(value),
+            AlertError::SendNotificationError { .. } => MetaHttpResponse::internal_error(value),
+            AlertError::GetDestinationWithTemplateError(error) => {
```

Comment:
ahh duh, we can just pass `value` directly, instead of each variant in the returns. TIL